### PR TITLE
Simplified Docker support.

### PR DESCRIPTION
**Pull request by petedyerarm**
Removed the scripts that generated and ran the images, instead this
information is in the README.md

Docker can be used for the wic-wic or the repo-repo generation scripts.

### CODE REVIEW

**Comment by JanneKiiskila** [comment]
@ygoyal18  @x85446  - please check, assign reviewers.

File: ostree/README.md [hunk 1]
```diff
@@ -14,33 +14,41 @@ Scripts are provided for both of these, and instructions for running directly or
...
 ```
-> ./createOSTreeUpdate <old-wic> <new-wic> [upgrade-tag]
+> ./createOSTreeUpdate.sh old-wic-file new-wic-file [upgrade-tag]
+```
+
+If using the Docker container use:
+
+```
+docker build --no-cache -f Docker/Dockerfile --label ostree-delta  --tag ${USER}/ostree-delta:latest .
+docker run --rm -v old_wic_file:/old_wic.wic -v new_wic_file:new_wic.wic -v /dev:/dev -v ${PWD}:/ws -w /ws --privileged ${USER}/ostree-delta:latest ./createOSTreeUpgrade.sh /old_wic.wic /new_wic.wic [upgrade-tag]
```
Comment (cmonr) [inline ostree/README.md:24]: Why is '--privileged' needed?

File: ostree/Docker/entrypoint.sh [hunk 2]
```diff
@@ -68,4 +68,5 @@ export HOME=/home/"$username"
 
 sed -i 's/%sudo  ALL=(ALL:ALL) ALL/%sudo ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers
 
-sudo -E -u "$username" "$@"
+"$@"
```
Comment (cmonr) [inline ostree/Docker/entrypoint.sh:71]: Since it looks like this runs as root, on the user's PWD, won't generated artifacts also be owned by root?

**Review by cmonr — COMMENTED** [review]
Had some questions.

File: ostree/README.md (hunk 1 above)
Comment (petedyerarm) [reply ostree/README.md:24]: Without it the mount of the loop device fails.
```
Mounting wic partition 2 of /tmp/tmp.jLdgG6Gizu/old_wic to /tmp/tmp.jLdgG6Gizu/old
mount: /tmp/tmp.jLdgG6Gizu/old: failed to setup loop device for /tmp/tmp.jLdgG6Gizu/old_wic.
```

File: ostree/Docker/entrypoint.sh (hunk 2 above)
Comment (petedyerarm) [reply ostree/Docker/entrypoint.sh:71]: Added instruction to use chown to fix this (similar to the snapcraft Quick start documentation).

**Review by x85446 — APPROVED** [review]

File: ostree/README.md (hunk 1 above)
Comment (JanneKiiskila) [reply ostree/README.md:24]: Might be good to comment that there?

Comment (petedyerarm) [reply ostree/README.md:24]: This PR has already been merged. 
Please re-check in https://github.com/PelionIoT/scripts-pelion-edge/pull/20

**Pull request merged**